python/ql/lib/semmle/python/dataflow/new/internal/VariableCapture.qll

```diff
@@ -29,7 +29,9 @@ private module CaptureInput implements Shared::InputSig<Location> {
   class BasicBlock extends PY::BasicBlock {
     Callable getEnclosingCallable() { result = this.getScope() }
 
-    // TODO: check that this gives useful results
+    // Note `PY:BasicBlock` does not have a `GetLocation` and
+    // has a more complicated location info logic.
+    // This is good enough for here.
```

Comment:
> This is good enough for here.

I don't understand, what does `This` refer to? and what does `here` refer to?

I gather it's something along these lines: 

> Using location of first node in the basic block is good enough, since a BasicBlock is only used for internal debugging purposes.

But I don't understand we wouldn't just use the location from the `PY::BasicBlock` though?

Reply:
> But I don't understand we wouldn't just use the location from the PY::BasicBlock though?

Because it doesn't have one.

Reply:
I tried to make the comment clearer.

Reply:
thanks :heart: